pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -122,7 +123,51 @@ func (bw *quotaRestrictedBlobWriter) Commit(ctx context.Context, provisional dis
 
 // admitBlobWrite checks whether the blob does not exceed image quota, if set. Returns
 // ErrAccessDenied error if the quota is exceeded.
-func admitBlobWrite(ctx context.Context, repo *repository) error {
+func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
+	if size < 1 {
+		return nil
+	}
+
+	err := admitLimitRanges(ctx, repo, size)
+	if err != nil {
+		return err
+	}
+
+	err = admitQuotas(ctx, repo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// admitLimitRanges checks the blob against any established limit ranges.
+func admitLimitRanges(ctx context.Context, repo *repository, size int64) error {
+	limitranges, err := repo.limitClient.LimitRanges(repo.namespace).List(kapi.ListOptions{})
+	if err != nil {
+		if kerrors.IsForbidden(err) {
+			context.GetLogger(ctx).Warnf("Cannot list limitranges because of outdated cluster roles: %v", err)
```

Comment:
Hmm.... I'm not sure about this approach. This is bypassing security, because cluster roles were not updated. But swallowing error here only to show it in logs won't help administrator notice the problem. I'd rather see users complaining about this and notify administrator of the problem. Otherwise, administrator will notice that only after users will start overusing assigned quotas.